python/mlx/nn/layers/activations.py

```diff
@@ -40,6 +41,19 @@ def leaky_relu(x, negative_slope=0.01):
     return mx.maximum(negative_slope * x, x)
 
 
+def log_softmax(x, axis=-1):
+    """Applies the Log Softmax function.
+
+    Applies :math:`\log(\frac{e^{x_i}}{\sum_j e^{x_j}})` element wise.
```

Comment:
I would do this instead as it more accurately reflects the actual computation and it's cleaner:

x + \log \sum_i e^{x_i}